daml-foundations/daml-ghc/daml-prim-src/GHC/Num.daml

```diff
@@ -63,6 +70,14 @@ class Additive a  where
 class Multiplicative a where
     (*) : a -> a -> a
     munit : a
+    -- | `x ^ n` raises `x` to the power of `n`.
+    (^): a -> Int -> a
```

Comment:
I'd be tempted not to give a default definition. It seems odd to have a default that is partial, but all useful implementations are total.

Reply:
The LF-native `Int` definition is partial. Indeed, for `Multiplicative`, there is only a partial definition, but one that i consider useful. If we want to have a default complete definition, I can move it to `Fractional`, but then the `Int` definition falls out of line as it errors on negative exponents rather than giving the expected result `0`.

Reply:
My point is why have a default definition at all? There's clearly variance over what it should do, which is fine, and nothing actually uses the default definition, so let's omit it.